### PR TITLE
feat(bottlecap): add `runtime` tag

### PR DESCRIPTION
**Pull request by duncanista**
# What?

Adds the `runtime` tag to the invocation span and metric tags.

Screenshot 2024-11-21 at 10 24 06 AM

Screenshot 2024-11-21 at 10 24 36 AM

### CODE REVIEW

File: bottlecap/src/lifecycle/invocation/processor.rs [hunk 1]
```diff
@@ -174,6 +185,13 @@ impl Processor {
                 proactive_initialization.to_string(),
             );
         }
+
+        if let Some(runtime) = &self.runtime {
+            self.span
+                .meta
+                .insert(String::from("runtime"), runtime.to_string());
+        }
```
Comment (duncanista) [inline bottlecap/src/lifecycle/invocation/processor.rs:193]: Not sure if we actually need this, it looks like previous versions have the `runtime` tag in spans, but not sure if this is done in backend or tracer.

File: bottlecap/src/lifecycle/invocation/processor.rs [hunk 2]
```diff
@@ -166,6 +172,11 @@ impl Processor {
             } else {
                 cold_start = true;
             }
+
+            // Resolve runtime only once
+            let runtime = resolve_runtime_from_proc(PROC_PATH, ETC_PATH);
+            self.enhanced_metrics.set_runtime_tag(&runtime);
```
Comment (duncanista) [inline bottlecap/src/lifecycle/invocation/processor.rs:178]: This is definitely needed tho

File: bottlecap/src/tags/lambda/tags.rs [hunk 3]
```diff
@@ -145,7 +145,7 @@ fn tags_from_env(
 }
 
 #[allow(dead_code)] // keeping this logic for when async runtime resolution will be supported
```
Comment (alexgallotta) [inline bottlecap/src/tags/lambda/tags.rs:147]: we can remove this now, right?

Comment (duncanista) [reply bottlecap/src/tags/lambda/tags.rs:147]: yeah

File: bottlecap/src/lifecycle/invocation/processor.rs (hunk 1 above)
Comment (duncanista) [reply bottlecap/src/lifecycle/invocation/processor.rs:193]: We do need it

**Pull request merged**